ts/main.ts

```diff
@@ -189,36 +208,70 @@ function clearCardList(): void {
   }
 }
 
-function populateExerciseDetails(baseId: number): void {
+function findExerciseByBaseId(baseId: number): Exercise | null {
   for (const exercise of exerciseObjArr) {
     if (exercise.baseId === baseId) {
-      $detailsTitle.innerHTML =
-        exercise.name +
-        ' <i class="fa-regular fa-heart" style="color: #ffc300"></i>';
-      $detailsImg.setAttribute('src', exercise.image);
-      if (exercise.primaryMuscles.length > 0) {
-        $detailsMusclePrim.textContent = '';
-        for (const muscle of exercise.primaryMuscles) {
-          $detailsMusclePrim.textContent += `${muscle.name}, `;
-        }
-      }
-      if (exercise.secondaryMuscles.length > 0) {
-        $detailsMuscleSec.textContent = '';
-        for (const muscle of exercise.secondaryMuscles) {
-          $detailsMuscleSec.textContent += `${muscle.name}, `;
-        }
-      }
-      if (exercise.equipment.length > 0) {
-        $detailsEquipment.textContent = '';
-        for (const equipment of exercise.equipment) {
-          $detailsEquipment.textContent += `${equipment.name}, `;
-        }
+      return exercise;
+    }
+  }
+  return null;
+}
+
+function handleFavoriteClick(
+  exerciseObj: Exercise,
+  targetIcon: HTMLElement,
+): void {
+  if (targetIcon.classList.contains('fa-regular')) {
+    targetIcon.classList.remove('fa-regular');
+    targetIcon.classList.add('fa-solid');
+    fitlogData.favorites.push(exerciseObj);
+    exerciseObj.favorite = true;
+  } else if (targetIcon.classList.contains('fa-solid')) {
+    targetIcon.classList.remove('fa-solid');
+    targetIcon.classList.add('fa-regular');
+    exerciseObj.favorite = false;
+    let indexToRemove = -1;
+    for (let i = 0; i < fitlogData.favorites.length; i++) {
+      if (fitlogData.favorites[i].baseId === exerciseObj.baseId) {
+        indexToRemove = i;
       }
-      $detailsDescription.innerHTML = exercise.description;
     }
+    fitlogData.favorites.splice(indexToRemove, 1);
   }
 }
 
+function populateExerciseDetails(baseId: number): void {
+  $exerciseDetailSection.setAttribute('data-base-id', `${baseId}`);
+  const exercise = findExerciseByBaseId(baseId);
+  if (!exercise) return;
+  $detailsTitle.textContent = exercise.name + ' ';
+  const $heart = document.createElement('i');
+  if (exercise.favorite) $heart.setAttribute('class', 'fa-solid fa-heart');
+  if (!exercise.favorite) $heart.setAttribute('class', 'fa-regular fa-heart');
+  $heart.setAttribute('style', 'color: #FFC300;');
+  $detailsTitle.appendChild($heart);
```

Comment:
If this is a `populate` function, why render the heart here? Can't you have the heart in your HTML, have queried for it and then determine whether or not it needs to be solid or not?

Reply:
```html
<h1 class="details-title">
          Exercise Name
          <i class="fa-regular fa-heart" style="color: #ffc300"></i>
        </h1>
```

I appended the heart as a Child so it would be apart of the heading element, so I have to render the heart there or else when I update the textContent of my details-title heading then the heart will disappear. Should I change how my HTML is formated for the details section?

Reply:
Yes, your heart icon shouldn't be a child of your `h1`. The only content that should be in your `h1` is text content.